Sources/Sentry/SentrySystemWrapper.mm

```diff
@@ -0,0 +1,117 @@
+#import "SentrySystemWrapper.h"
+#import "SentryDispatchSourceWrapper.h"
+#import "SentryError.h"
+#import "SentryMachLogging.hpp"
+#import <mach/mach.h>
+
+const NSUInteger kSentryMemoryPressureLevelNormal = DISPATCH_MEMORYPRESSURE_NORMAL;
+const NSUInteger kSentryMemoryPressureLevelWarn = DISPATCH_MEMORYPRESSURE_WARN;
+const NSUInteger kSentryMemoryPressureLevelCritical = DISPATCH_MEMORYPRESSURE_CRITICAL;
+
+@implementation SentrySystemWrapper {
+    dispatch_queue_t _memoryWarningQueue;
+    SentryDispatchSourceFactory *_dispatchSourceFactory;
+    SentryDispatchSourceWrapper *_dispatchSourceWrapper;
+}
+
+- (instancetype)initWithDispatchSourceFactory:(SentryDispatchSourceFactory *)dispatchSourceFactory
+{
+    if (self = [super init]) {
+        _dispatchSourceFactory = dispatchSourceFactory;
+
+        const auto queueAttributes
+            = dispatch_queue_attr_make_with_qos_class(DISPATCH_QUEUE_SERIAL, QOS_CLASS_UTILITY, 0);
+        _memoryWarningQueue
+            = dispatch_queue_create("io.sentry.queue.memory-warnings", queueAttributes);
+    }
+    return self;
+}
+
+- (SentryRAMBytes)memoryFootprintBytes:(NSError *__autoreleasing _Nullable *)error
+{
+    task_vm_info_data_t info;
+    mach_msg_type_number_t count = TASK_VM_INFO_COUNT;
+
+    const auto status = task_info(mach_task_self(), TASK_VM_INFO, (task_info_t)&info, &count);
+    if (status != KERN_SUCCESS) {
+        if (error) {
+            *error = NSErrorFromSentryErrorWithKernelError(
+                kSentryErrorKernel, @"task_info reported an error.", status);
+        }
+        return 0;
+    }
+
+    SentryRAMBytes footprintBytes;
+    if (count >= TASK_VM_INFO_REV1_COUNT) {
+        footprintBytes = info.phys_footprint;
+    } else {
+        footprintBytes = info.resident_size;
+    }
+
+    return footprintBytes;
+}
+
+- (NSArray<NSNumber *> *)cpuUsagePerCore:(NSError **)error
```

Comment:
`l`: There should be a linter for this.
```suggestion
- (nullable NSArray<NSNumber *> *)cpuUsagePerCore:(NSError **)error
```

Reply:
Maybe. I typically only add nullability annotations for public methods to their declarations in header files.